L1Trigger/TrackFindingTMTT/interface/Array2D.h

```diff
@@ -0,0 +1,34 @@
+#ifndef L1Trigger_TrackFindingTMTT_Array2D_h
+#define L1Trigger_TrackFindingTMTT_Array2D_h
+
+#include <vector>
+
+//=== Generic 2D array class.
+
+// (Replaced boost::numeric::ublas::matrix when boost library
+//  became too big).
+
+// Author: Lucas Camolezi
+
+namespace tmtt {
+
+  template <typename T>
+  class Array2D {
+  public:
+    //for a mxn matrix - row major
+    Array2D(unsigned int m, unsigned int n) : _n{n}, _m{m} { array2D_.resize(m * n); }
+    const T& operator()(unsigned int i, unsigned int j) const { return array2D_.at(i * _n + j); }
```

Comment:
1) Bound checking now included in both const & non-const operator, with bound checking implemented in new private function called from both these original operators, to minimise duplicate code. (Duplicate code could be eliminated altogether if you prefer, using the "const_cast" solution of Scott Meyers https://stackoverflow.com/questions/123758/how-do-i-remove-code-duplication-between-similar-const-and-non-const-member-func , but doesn't seem worth it for so little code).
2) Const operator must call array2D_.at() whereas non-const operator can't do this, so this difference remains.
3) It's true that having both Array2D::() and Array2D::at() functions would be more general. But L1 track development code will always want array bound checking on.

Reply:
> Const operator must call array2D_.at() whereas non-const operator can't do this, so this difference remains.

I don't understand this claim. For `std::vector`, both `at()` and `operator[]` provide const and non-const versions. Since the bounds checking is already applied in a separate function, it's most efficient to use `operator[]` rather than `at()` everywhere (because `at()` checks bounds again, but in a 1D fashion rather than the 2D fashion of `checkBounds()`, therefore both redundant and less useful).

Reply:
You're right. I changed this again. Internally, Array2D now uses std::vector::operator[] so no std bound checking is done. Since both const & non-const member functions would have the same code, the C++17 version of Scott Meyers solution above was used to avoid duplication. The checkBounds() function has been deleted, as no longer needed.